fishtest/fishtest/rundb.py

```diff
@@ -277,10 +277,9 @@ def flush_all(self):
         # Note that we do not grab locks because this method is
         # called from a signal handler and grabbing locks might deadlock
         for r_id in list(self.run_cache):
-            if r_id in self.run_cache and self.run_cache[r_id]["dirty"]:
-                self.runs.replace_one(
-                    {"_id": ObjectId(r_id)}, self.run_cache[r_id]["run"]
-                )
+            entry = self.run_cache.get(r_id, None)
+            if entry is not None and entry["dirty"]:
```

Comment:
If `entry = self.run_cache.get(r_id, None)` returns dict/None (with `dirty` key mandatory in dict) then should work also the shorter:
```python
    if entry and entry["dirty"]:
```

Reply:
`rundb.py` has a lot `if X is not none` at other places. We should replace them all to be consistent. I am not sure what is considered the most elegant Pythonic way.

Reply:
I prefer the riskier `if X:` with only one change, the safer `if X is not none` with several changes :)
Anyway they are very different, consider the case of `entry = {}` (a bug somewhere) and decide what the code should do in that case.

Reply:
Google Python Style Guide
https://google.github.io/styleguide/pyguide.html#214-truefalse-evaluations

Use the “implicit” false if possible, e.g., `if foo:` rather than `if foo != []:`.
There are a few caveats that you should keep in mind though:
* Always use `if foo is None:` (or `is not None`) to check for a `None` value. E.g., when testing whether a variable or argument that defaults to `None` was set to some other value. The other value might be a value that’s false in a boolean context!

Reply:
Ok, so I think I prefer the current "is not None" as hinted by:

> Always use if foo is None: (or is not None) to check for a None value